processors/src/main/java/io/opentelemetry/contrib/eventbridge/EventToSpanEventBridge.java

```diff
@@ -0,0 +1,136 @@
+ * events for the current span if:
+ *
+ * <ul>
+ *   <li>The log record has a valid span context
+ *   <li>{@link Span#current()} returns a span where {@link Span#isRecording()} is true
+ *   <li>The log record's span context is the same as {@link Span#current()}
+ * </ul>
+ *
+ * <p>The event {@link LogRecordData} is converted to a span event as follows:
+ *
+ * <ul>
+ *   <li>{@code event.name} attribute is mapped to span event name
+ *   <li>{@link LogRecordData#getTimestampEpochNanos()} is mapped to span event timestamp
+ *   <li>{@link LogRecordData#getAttributes()} are mapped to span event attributes, excluding {@code
+ *       event.name}
+ *   <li>{@link LogRecordData#getObservedTimestampEpochNanos()} is mapped to span event attribute
+ *       with key {@code log.record.observed_timestamp}
+ *   <li>{@link LogRecordData#getSeverity()} is mapped to span event attribute with key {@code
+ *       log.record.severity_number}
+ *   <li>{@link LogRecordData#getBodyValue()} is mapped to span event attribute with key {@code
+ *       log.record.body}, as an escaped JSON string following the standard protobuf JSON encoding
+ *   <li>{@link LogRecordData#getTotalAttributeCount()} - {@link
+ *       LogRecordData#getAttributes()}.size() is mapped to span event attribute with key {@code
+ *       log.record.dropped_attributes_count}
+ * </ul>
+ */
+public final class EventToSpanEventBridge implements LogRecordProcessor {
+
+  private static final Logger logger = Logger.getLogger(EventToSpanEventBridge.class.getName());
+
+  private static final AttributeKey<String> EVENT_NAME = AttributeKey.stringKey("event.name");
+  private static final AttributeKey<Long> LOG_RECORD_OBSERVED_TIME_UNIX_NANO =
+      AttributeKey.longKey("log.record.observed_time_unix_nano");
+  private static final AttributeKey<Long> LOG_RECORD_SEVERITY_NUMBER =
+      AttributeKey.longKey("log.record.severity_number");
+  private static final AttributeKey<String> LOG_RECORD_BODY =
+      AttributeKey.stringKey("log.record.body");
+  private static final AttributeKey<Long> LOG_RECORD_DROPPED_ATTRIBUTES_COUNT =
+      AttributeKey.longKey("log.record.dropped_attributes_count");
+
+  private EventToSpanEventBridge() {}
+
+  /** Create an instance. */
+  public static EventToSpanEventBridge create() {
+    return new EventToSpanEventBridge();
+  }
+
+  @Override
+  public void onEmit(Context context, ReadWriteLogRecord logRecord) {
+    LogRecordData logRecordData = logRecord.toLogRecordData();
+    String eventName = logRecordData.getAttributes().get(EVENT_NAME);
+    if (eventName == null) {
+      return;
+    }
+    SpanContext logSpanContext = logRecordData.getSpanContext();
+    if (!logSpanContext.isValid()) {
+      return;
+    }
+    Span currentSpan = Span.current();
+    if (!currentSpan.isRecording()) {
+      return;
+    }
+    if (!currentSpan.getSpanContext().equals(logSpanContext)) {
+      return;
+    }
+    currentSpan.addEvent(
```

Comment:
@jack-berg I just found this in the otep about events

https://github.com/open-telemetry/oteps/blob/main/text/0265-event-vision.md#relationship-to-span-events

It says span events will be deprecated in the API. I suppose it will still stay in SDK / proto because many backends only support trace signal. Is there a non-API approach for this processor? Or is it not worth reading too much into that for now?

Reply:
the purpose of deprecating won't be to remove, but rather to clearly signal when it is no longer the preferred approach

I don't think we will ever remove it since there's no plan for breaking changes